src/helpers/HookConfigLib.sol

```diff
@@ -0,0 +1,140 @@
+// 0x________________________________________BBBBBBBB________________ // Entity ID
+// 0x________________________________________________CC______________ // Type
+// 0x__________________________________________________DD____________ // exec hook flags
+//
+
+// Hook types:
+// 0x00 // Exec
+// 0x01 // Validation
+
+// Exec hook flags layout:
+// 0b000000__ // unused
+// 0b______A_ // hasPre
+// 0b_______B // hasPost
+
+library HookConfigLib {
+    // Hook type constants
+    // Exec has no bits set
+    bytes32 internal constant _HOOK_TYPE_EXEC = bytes32(uint256(0));
+    // Validation has 1 in the 25th byte
+    bytes32 internal constant _HOOK_TYPE_VALIDATION = bytes32(uint256(1) << 56);
+
+    // Exec hook flags constants
+    // Pre hook has 1 in 2's bit in the 26th byte
+    bytes32 internal constant _EXEC_HOOK_HAS_PRE = bytes32(uint256(1) << 49);
+    // Post hook has 1 in 1's bit in the 26th byte
+    bytes32 internal constant _EXEC_HOOK_HAS_POST = bytes32(uint256(1) << 48);
+
+    function packValidationHook(ModuleEntity _hookFunction) internal pure returns (HookConfig) {
+        return
+            HookConfig.wrap(bytes26(bytes26(ModuleEntity.unwrap(_hookFunction)) | bytes26(_HOOK_TYPE_VALIDATION)));
+    }
+
+    function packValidationHook(address _module, uint32 _entityId) internal pure returns (HookConfig) {
+        return HookConfig.wrap(
+            bytes25(
+                // module address stored in the first 20 bytes
+                bytes25(bytes20(_module))
+                // entityId stored in the 21st - 24th byte
+                | bytes25(bytes24(uint192(_entityId))) | bytes25(_HOOK_TYPE_VALIDATION)
+            )
+        );
+    }
+
+    function packExecHook(ModuleEntity _hookFunction, bool _hasPre, bool _hasPost)
+        internal
+        pure
+        returns (HookConfig)
+    {
+        return HookConfig.wrap(
+            bytes26(
+                bytes26(ModuleEntity.unwrap(_hookFunction))
+                // | bytes26(_HOOK_TYPE_EXEC) // Can omit because exec type is 0
+                | bytes26(_hasPre ? _EXEC_HOOK_HAS_PRE : bytes32(0))
+                    | bytes26(_hasPost ? _EXEC_HOOK_HAS_POST : bytes32(0))
+            )
+        );
+    }
+
+    function packExecHook(address _module, uint32 _entityId, bool _hasPre, bool _hasPost)
+        internal
+        pure
+        returns (HookConfig)
+    {
+        return HookConfig.wrap(
+            bytes26(
+                // module address stored in the first 20 bytes
+                bytes26(bytes20(_module))
+                // entityId stored in the 21st - 24th byte
+                | bytes26(bytes24(uint192(_entityId)))
+                // | bytes26(_HOOK_TYPE_EXEC) // Can omit because exec type is 0
+                | bytes26(_hasPre ? _EXEC_HOOK_HAS_PRE : bytes32(0))
+                    | bytes26(_hasPost ? _EXEC_HOOK_HAS_POST : bytes32(0))
+            )
+        );
+    }
```

Comment:
Do we need this function? Seems duplicated code. 
This can be achieved through the above function:
```
packExecHook(ModuleEntityLib.pack(module, entityId), hasPre, hasPost)
```

Reply:
Yeah It's a bit of a duplicate, but it's just for convenience to the caller so they don't have to import two libraries in some cases. I think this function is only linked in some tests, so it doesn't increase the bytecode size for the actual contracts.